nexus/db-queries/src/db/datastore/zpool.rs

```diff
@@ -32,6 +33,41 @@ use omicron_common::api::external::ResourceType;
 use uuid::Uuid;
 
 impl DataStore {
+    // TODO: Should this be paginated?
+    // TODO: Alternatively, should we have a better way of "inferring what needs
+    // to be updated" from inventory?
+    pub async fn zpool_update_all(
```

Comment:
I feel a little uneasy about this invocation.

Basically, we have a big set of "all zpools" from inventory, and a big set of "all zpools" in the DB we're using as "control plane objects". This query updates all the live zpools using information from inventory, all at once.

I could see this as useful in the future for flagging zpools with "high" usage (regardless of their "provisioned" usage), or zpools that have faulted have have a state other than `ONLINE`. 

However, for just setting `total_size`, a value which **should** only be set once, it feels a little heavy-handed?

I think I'd only want to move forward with this PR if we think this other "dynamic zpool information informing database data" pattern seems useful generally.

Reply:
I went ahead and removed this, instead, we're joining with the latest `inv_zpool` row.